node/bft/ledger-service/src/ledger.rs

```diff
@@ -127,20 +134,30 @@ impl<N: Network, C: ConsensusStorage<N>> LedgerService<N> for CoreLedgerService<
     /// Returns the committee for the given round.
     /// If the given round is in the future, then the current committee is returned.
     fn get_committee_for_round(&self, round: u64) -> Result<Committee<N>> {
-        match self.ledger.get_committee_for_round(round)? {
+        // Check if the committee is already in the cache.
+        if let Some(committee) = self.committee_cache.lock().get(&round) {
+            return Ok(committee.clone());
+        }
+
+        let committee = match self.ledger.get_committee_for_round(round)? {
             // Return the committee if it exists.
-            Some(committee) => Ok(committee),
+            Some(committee) => committee,
             // Return the current committee if the round is in the future.
             None => {
                 // Retrieve the current committee.
                 let current_committee = self.current_committee()?;
                 // Return the current committee if the round is in the future.
                 match current_committee.starting_round() <= round {
-                    true => Ok(current_committee),
+                    true => current_committee,
                     false => bail!("No committee found for round {round} in the ledger"),
                 }
             }
-        }
+        };
+
+        // Insert the committee into the cache.
+        self.committee_cache.lock().push(round, committee.clone());
```

Comment:
@ljedrz this only works for the `match` case on `Some`. It does not work for the committee from `None`.

Can you scope the cache for only committees from `Some` and profile the performance again?

(If you want to go the extra mile, you can also try adding a `cache` for the underlying impl of `self.current_committee()`, but this extra overhead may not be worth the complexity. The majority of calls will route to `Some` I believe)

Reply:
Fixed; I ran the branch locally and all I can see are still cache hits. As for the current committee, I agree that its overhead is low; we can always go back to that idea in the future if need be.